frontend/src/scenes/plugins/edit/PluginDrawer.tsx

```diff
@@ -69,7 +69,7 @@ export function PluginDrawer(): JSX.Element {
         } else {
             form.resetFields()
         }
-    }, [editingPlugin?.id])
+    }, [editingPlugin?.id, editingPlugin?.config_schema])
```

Comment:
@mariusandra  - based on my 30 minute crash course on Kea, this _feels_ a bit wrong. Shouldn't there be a `listener` for the `editingPlugin` action which sets the form fields?

I'm not quite sure how the `form` would get passed there, but I spent waay too long trying to figure out where in the logic to find the offending piece.

Reply:
I think this is OK, that is passing a form into Kea feels even wronger to me, and there's no other way to affect it in a listener.

Reply:
Hmm, I'm guessing because form is a frontend component, it shouldn't make it into the logic ever? <- I can get behind this.

Reply:
Actually, `generateApiKeysIfNeeded` action takes the `form` as a parameter, too. 🤔

Reply:
Yeah, `antd` forms try to push us towards a very different way of handling data than what Kea is used to. This `useEffect` is a way to somewhat tame the two. I don't know what's the best way forward for these changes. It's all rather flimsy, so make sure to test test test.

I do think that frontend forms should probably also follow a data-first mindset (rather than being directed by the template engine aka React), and will eventually look like a modern version of https://kea-v1.netlify.app/guide/forms -- I'm still unsure of the exact API and if it should be a kea plugin or just a function wrapper like in that link.